packages/tldraw/src/lib/ui/context/actions.tsx

```diff
@@ -261,29 +261,31 @@ export function ActionsProvider({ overrides, children }: ActionsProviderProps) {
 
 					trackEvent('toggle-auto-size', { source })
 					editor.markHistoryStoppingPoint('toggling auto size')
-					const shapes = editor
-						.getSelectedShapes()
-						.filter(
-							(shape): shape is TLTextShape =>
-								editor.isShapeOfType<TLTextShape>(shape, 'text') && shape.props.autoSize === false
+					editor.run(() => {
```

Comment:
not for this PR, but i think it would be good for DX to have `editor.run` take a history label (in this case, "toggling auto size") as part of the options. `editor.run` is basically a wrapper around history, so it would make sense to have it do `markHistoryStoppingPoint` under the hood. otherwise, people have to remember to do two separate things to make this work correctly.

Reply:
I'm almost certain we had this at one point, where editor.mark and ye olde editor.batch were the same thing. There were so many cases of needing to have multiple frames-worth of stuff on a single undo/redo entry that the option felt like a trap so we separated it out when we refactored history.